pc2cps

```diff
@@ -72,18 +86,60 @@ class pkgconfig(object):
             else:
                 deps[dep[0]] = {}
 
-        return deps
+        self.requires = deps
 
     #--------------------------------------------------------------------------
     @property
     def component(self):
         def _keys(x):
-            return None if x is None else x.keys()
-
-        c = {
-            'Type': 'interface',
-        }
-        extend(c, 'Link-Flags', self._get_split('Libs'))
+            return [] if x is None else x.keys()
+
+        def _list(re_):
+            ma = [re_.match(x) for x in link_flags]
+            return [x.group(1) for x in ma if x]
+
+        # If not found and if package name given in library list with
+        # "-l" flag, an error should be thrown?
+        def _find_location():
+            for path in self._paths:
+                for file in os.listdir(path):
+                    pattern = '(lib)?' + self.library_name + self._lib_pattern
+                    ma = re.match(pattern, file)
+                    if ma is not None:
+                        return os.path.join(path, ma.group(0))
```

Comment:
I think this is missing logic to deal with the case of both shared and static libraries matching. (Um... to *properly* handle that would require being aware of the linker flags that control which library the linker will choose in such cases... maybe we don't need to go *that* far, although a `TODO` might be in order, but it seems like we should at least implement our own preference, rather than relying on the order in which files are iterated?)

Actually, I wonder if it wouldn't be better to approach this entire problem as a state machine; that is, parse each link flag one at a time and take appropriate action, which would be either modifying the parse state, locating a library to add to the list of link libraries, or passing along a link flag. (This also makes me wonder if we ever want to pass along the link flags to change the linker's static/shared preference...)

Reply:
Indeed, I could add a flag to specified the preferred order in which the type of the library (static/shared) is chosen.

At first, when I started to work on this PR, I was searching for all the libraries that were in the linker flags. But since `pc` files are, as far as I know, only declare one component/target, I thought it would make sense to convert the libraries found in the flags to `required` components instead of looking for their paths.